src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
@lyptt I'll move myself the code, then you will have to `git rebase` in order to update your branch